src/java/htsjdk/samtools/metrics/MetricsFile.java

```diff
@@ -267,10 +274,9 @@ private void printHistogram(final BufferedWriter out, final FormatUtil formatter
 
             for (final Histogram<HKEY> histo : nonEmptyHistograms) {
                 final Histogram<HKEY>.Bin bin = histo.get(key);
-                final double value = (bin == null ? 0 : bin.getValue());
-
                 out.append(SEPARATOR);
-                out.append(formatter.format(value));
+
+                if(bin != null) out.append(formatter.format(bin.getValue()));
```

Comment:
Does this refactoring change the behavior?  Did there used to be 0's output that are now skipped?